flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/api/java/internal/StreamTableEnvironmentImpl.java

```diff
@@ -357,6 +361,26 @@ public StreamExecutionEnvironment execEnv() {
 		return executionEnvironment;
 	}
 
+	/**
+	 * This method is used for sql client to submit job.
+	 */
+	public Pipeline getPipeline(String jobName) {
+		return execEnv.createPipeline(translate(), tableConfig, jobName);
+	}
+
+	@Override
+	protected List<Transformation<?>> translate(List<ModifyOperation> modifyOperations) {
+		// keep the behavior as before: translate each operation independently
+		List<Transformation<?>> transformations = new ArrayList<>(modifyOperations.size());
+		for (ModifyOperation operation : modifyOperations) {
+			List<Transformation<?>> transformationList = planner.translate(Collections.singletonList(operation));
+			Preconditions.checkArgument(transformationList.size() == 1,
+				"expected size is 1, actual size is " + transformationList.size());
+			transformations.add(transformationList.get(0));
+		}
```

Comment:
agree with you. we disable multi-sink optimization in `StreamTableEnvironmentImpl` before because `StreamExecutionEnvironment` can also trigger the table program (generated by `StreamTableEnvironmentImpl`) and DataStream program. Now `StreamExecutionEnvironment` can only trigger DataStream program, `StreamTableEnvironmentImpl ` can also support multi-sink optimization. `StreamTableEnvironmentImpl`  supports full functionality, including all kinds of UDFs registration, to/from DataStream, multi-sink optimization.